core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala

```diff
@@ -245,27 +208,18 @@ private[spark] object HadoopFSUtils extends Logging {
         Array.empty[FileStatus]
     }
 
-    def doFilter(statuses: Array[FileStatus]) = filterFun match {
-      case Some(shouldFilterOut) =>
-        statuses.filterNot(status => shouldFilterOut(status.getPath.getName))
-      case None =>
-        statuses
-    }
-
-    val filteredStatuses = doFilter(statuses)
     val allLeafStatuses = {
-      val (dirs, topLevelFiles) = filteredStatuses.partition(_.isDirectory)
+      val (dirs, topLevelFiles) = statuses.partition(_.isDirectory)
```

Comment:
@sunchao the `dirs` here may contain hidden directories. We still need to filter them before listing leaf files.

Reply:
@gengliangwang you're right. Thanks for catching this! and sorry for introducing this regression.